components/scream/src/share/io/tests/io.cpp

```diff
@@ -0,0 +1,294 @@
+  // 1 file each for averaged, instantaneous, min and max data.
+  // Cycle through each output and make sure it is correct.
+  // We can use the produced output files to simultaneously check output quality and the
+  // ability to read input.
+  auto ins_params = get_in_params("Instant",io_comm);
+  auto avg_params = get_in_params("Average",io_comm);
+  auto min_params = get_in_params("Min",io_comm);
+  auto max_params = get_in_params("Max",io_comm);
+  Real tol = pow(10,-6);
+  // Check instant output
+  input_type ins_input(io_comm,ins_params,field_repo,grid_man);
+  ins_input.pull_input();
+  auto f1_dev = field_repo->get_field("field_1","Physics").get_view();
+  auto f2_dev = field_repo->get_field("field_2","Physics").get_view();
+  auto f3_dev = field_repo->get_field("field_3","Physics").get_reshaped_view<Real**>();
+  auto f1_hst = Kokkos::create_mirror_view( f1_dev );
+  auto f2_hst = Kokkos::create_mirror_view( f2_dev );
+  auto f3_hst = Kokkos::create_mirror_view( f3_dev );
+  Kokkos::deep_copy( f1_hst, f1_dev );
+  Kokkos::deep_copy( f2_hst, f2_dev );
+  Kokkos::deep_copy( f3_hst, f3_dev );
+  for (int ii=0;ii<num_lcols;++ii)
+  {
+    REQUIRE(std::abs(f1_hst(ii)-(max_steps*dt+ii))<tol);
+    for (int jj=0;jj<num_levs;++jj)
+    {
+      REQUIRE(std::abs(f3_hst(ii,jj)-(ii+max_steps*dt + (jj+1)/10.))<tol);
+    }
+  }
+  for (int jj=0;jj<num_levs;++jj)
+  {
+    REQUIRE(std::abs(f2_hst(jj)-(max_steps*dt + (jj+1)/10.))<tol);
+  }
+  // Check average output
+  input_type avg_input(io_comm,avg_params,field_repo,grid_man);
+  avg_input.pull_input();
+  Kokkos::deep_copy( f1_hst, f1_dev );
+  Kokkos::deep_copy( f2_hst, f2_dev );
+  Kokkos::deep_copy( f3_hst, f3_dev );
+  Real avg_val;
+  for (int ii=0;ii<num_lcols;++ii)
+  {
+    avg_val = (max_steps+1)/2.0*dt + ii; // Sum(x0+i*dt,i=1...N) = N*x0 + dt*N*(N+1)/2, AVG = Sum/N, note x0=ii in this case
+    REQUIRE(std::abs(f1_hst(ii)-avg_val)<tol);
+    for (int jj=0;jj<num_levs;++jj)
+    {
+      avg_val = (max_steps+1)/2.0*dt + (jj+1)/10.+ii;  //note x0=(jj+1)/10+ii in this case.
+      REQUIRE(std::abs(f3_hst(ii,jj)-avg_val)<tol);
+    }
+  }
+  for (int jj=0;jj<num_levs;++jj)
+  {
+    avg_val = (max_steps+1)/2.0*dt + (jj+1)/10.;  //note x0=(jj+1)/10 in this case.
+    REQUIRE(std::abs(f2_hst(jj)-avg_val)<tol);
+  }
+  // Check max output
+  // The max should be equivalent to the instantaneous because this function is monotonically increasing.
+  input_type max_input(io_comm,max_params,field_repo,grid_man);
+  max_input.pull_input();
+  Kokkos::deep_copy( f1_hst, f1_dev );
+  Kokkos::deep_copy( f2_hst, f2_dev );
+  Kokkos::deep_copy( f3_hst, f3_dev );
+  for (int ii=0;ii<num_lcols;++ii)
+  {
+    REQUIRE(std::abs(f1_hst(ii)-(max_steps*dt+ii))<tol);
+    for (int jj=0;jj<num_levs;++jj)
+    {
+      REQUIRE(std::abs(f3_hst(ii,jj)-(ii+max_steps*dt + (jj+1)/10.))<tol);
+    }
+  }
+  for (int jj=0;jj<num_levs;++jj)
+  {
+    REQUIRE(std::abs(f2_hst(jj)-(max_steps*dt + (jj+1)/10.)<tol));
```

Comment:
My editor suggests that there might be an error here, since we're taking the abs of a bool. Should this perhaps be:

```
REQUIRE(std::abs(f2_hst(jj)-(max_steps*dt + (jj+1)/10.))<tol);
```
Notice the parentheses at the end: from `<tol))` to `)<tol)`.

Reply:
Yes, that change is correct.  Weird that you can take `std::abs(bool)` without throwing an error.  @bartgol , were you planning on pushing a version of the branch now that you may have figured out was making Blake go crazy?  If so, I can wait to put this change in.  Otherwise I'll push the fix this morning.

Reply:
nevermind, I just noticed that the AT is showing passing, so your fix must already be active and on the blake side of things.  I'll re-inspect the tests and make sure there are no other weird things like this.  Then push the changes.

Reply:
@bartgol , I pushed a fix for this error.